web/crux/src/app/product/guards/productId.validation.guard.ts

```diff
@@ -0,0 +1,46 @@
+import { BadRequestException, CanActivate, ExecutionContext, Injectable } from '@nestjs/common'
+import { Reflector } from '@nestjs/core'
+
+import validator from 'validator'
+
+/**
+ * UuidValidationGuard ensures that incoming requests contain valid UUIDs for
+ * specified parameters, and rejects the request if any of the UUIDs
+ * are invalid.
+ */
+@Injectable()
+export default class UuidValidationGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const params = this.reflector.get<string[]>('params', context.getHandler())
+    const req = context.switchToHttp().getRequest()
+
+    // return if there is no given params
+    if (!params) {
+      return true
+    }
+
+    let validUuid
+
+    params.forEach(actualParam => {
+      const id = req.params[actualParam] as string
+
+      if (!id) {
+        validUuid = true
+      }
+
+      if (!validator.isUUID(id)) {
+        throw new BadRequestException({
+          message: 'Given UUID is Invalid.',
+          property: actualParam,
+          value: id,
+        })
+      } else {
+        validUuid = true
+      }
+    })
+
+    return validUuid
```

Comment:
This is unnecessary. As we always throw an error on the first invalid uuid we can simply `return true` here.